### ISSUE TITLE
[Bug]: Failed copying a file because a directory exists at the target path

### ISSUE THREAD

**Issue by alexkuretz** [garden-io/garden#6456]
### Current behavior
Almost certain this is related to https://github.com/garden-io/garden/issues/2382 regarding symlink directories that was fixed in 0.13.40. If we downgrade to 0.13.31 (which is the version of our last release of this project) the builds succeed.

```
✖ build.service             → Failed processing Build type=container name=service (from module service) (took 21.26 sec). This is what happened:

────────────────────────────────────────────────────────────────────────────────────────────────────
Build staging: Failed copying file from '/home/runner/work/repo-name/repo-name/plugin/node_modules/.bin/myexecutable' to '/home/runner/work/repo-name/repo-name/.garden/build/plugin/node_modules/.bin/myexecutable' because a directory exists at the target path
```

The directory in question contains a slew of symlinks.
```
$ ll .garden/build/plugin/node_modules/.bin
total 24
drwxr-xr-x   2 myuser mygroup  4096 Sep 19 16:50 ./
drwxr-xr-x 493 myuser mygroup  20480 Sep 19 16:50 ../
lrwxrwxrwx   1 myuser mygroup  18 Sep 19 16:50 acorn -> ../acorn/bin/acorn*
lrwxrwxrwx   1 myuser mygroup  13 Sep 19 16:50 ava -> ../ava/cli.js*
lrwxrwxrwx   1 myuser mygroup  25 Sep 19 16:50 esparse -> ../esprima/bin/esparse.js*
```

### Expected behavior
Behavior prior to 0.13.40 works, but I also need the symlink dirs to work as was fixed in 0.13.40

### Workaround
Downgrade to garden 0.13.31, or probably any version prior to 0.13.40

### Suggested solution(s)
Unsure

### Your environment

* **OS:** Ubuntu
* **How I'm running Kubernetes:** EKS
Garden version: 0.13.40

**Comment by stefreak** [comment]
@alexkuretz Hi, I'm working on reproducing this bug now; Is there a full stack trace of this error in debug (`-l4`) or silly (`-l5`) loglevel?

**Comment by stefreak** [comment]
Successfully reproduced with this minimal example, when using `copyFrom`: 
[repro.tar.gz](https://github.com/user-attachments/files/17128832/repro.tar.gz)

```
apiVersion: garden.io/v1
kind: Project
name: test
defaultEnvironment: local
environments:
 - name: local
---
kind: Build
type: exec
name: test
spec:
  command: ["ls", "-lah", "node_modules/.bin"]
---
kind: Build
type: exec
name: test2
include: []
dependencies:
  - build.test
copyFrom:
  - build: test
    sourcePath: .
spec:
  command: ["ls", "-lah", "node_modules/.bin"]
```

```
repro % garden build test2 -l3
[verbose] garden version: 0.13.40
Build 🔨

ℹ garden                    → Initializing...
Unable to authenticate against the Garden dashboard with the current session token.
Command results for this command run will not be available in the Garden dashboard. If this not a
the Garden dashboard project you can ignore this warning. Otherwise, please try logging out with
`garden logout` and back in again with `garden login`.
ℹ garden-dashboard          → You are not logged in. To use the Garden dashboard, log in with the garden login command.
ℹ garden-dashboard          → Learn more at: https://docs.garden.io/using-garden/dashboard
🌿 Log in with garden login to explore logs, past commands, and your dependency graph in the Garden dashboard.
Learn more at: https://docs.garden.io/using-garden/dashboard

→ Run garden util hide-warning web-app to disable this message.
ℹ garden                    → Running in environment local.default
ℹ git                       → [verbose] Scanning project root at /Users/steffen/repro
  → Includes: (none)
  → Excludes: (none)
ℹ git                       → [verbose] Found 7 files in project root /Users/steffen/repro (took 0.05 sec)
ℹ providers                 → Resolving providers...
ℹ exec                      → Configuring provider...
✔ exec                      → Provider configured
✔ exec                      → Provider ready
ℹ providers                 → [verbose] resolve provider container is ready.
ℹ providers                 → [verbose] resolve provider exec is ready.
ℹ providers                 → [verbose] resolve provider templated is ready.
✔ providers                 → Finished resolving providers (took 0.1 sec)
ℹ graph                     → Resolving actions and modules...
ℹ git                       → [verbose] Scanning Build action test root at /Users/steffen/repro
  → Includes: (none)
  → Excludes: (none)
ℹ git                       → [verbose] Found 7 files in Build action test root /Users/steffen/repro (took 0.03 sec)
✔ graph                     → Finished resolving graph (took 0 sec)
ℹ garden                    → Finished initializing Garden (took 1 sec)

ℹ build.test2               → Getting status for Build test2 (type exec) at version v-5720dec677...
ℹ build.test2               → Status is in an unknown state, test2 will be built
ℹ build.test                → Getting status for Build test (type exec) at version v-8d53cb4cfe...
ℹ build.test                → Status is in an unknown state, test will be built
ℹ build.test                → Building test (type exec) at version v-8d53cb4cfe...
ℹ build.test                → [verbose] Syncing sources (6 files)...
ℹ build.test                → [verbose] Done syncing sources (took 0 sec)
ℹ build.test                → [verbose] [ls] total 0
ℹ build.test                → [verbose] [ls] drwxr-xr-x  5 steffen  staff   160B 25 Sep 11:44 .
ℹ build.test                → [verbose] [ls] drwxr-xr-x  6 steffen  staff   192B 25 Sep 11:44 ..
ℹ build.test                → [verbose] [ls] lrwxr-xr-x  1 steffen  staff    18B 25 Sep 11:44 acorn -> ../acorn/bin/acorn
ℹ build.test                → [verbose] [ls] lrwxr-xr-x  1 steffen  staff    13B 25 Sep 11:44 ava -> ../ava/cli.js
ℹ build.test                → [verbose] [ls] lrwxr-xr-x  1 steffen  staff    25B 25 Sep 11:44 esparse -> ../esprima/bin/esparse.js
ℹ build.test                → Finished building test. Here is the full output:

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
total 0
drwxr-xr-x  5 steffen  staff   160B 25 Sep 11:44 .
drwxr-xr-x  6 steffen  staff   192B 25 Sep 11:44 ..
lrwxr-xr-x  1 steffen  staff    18B 25 Sep 11:44 acorn -> ../acorn/bin/acorn
lrwxr-xr-x  1 steffen  staff    13B 25 Sep 11:44 ava -> ../ava/cli.js
lrwxr-xr-x  1 steffen  staff    25B 25 Sep 11:44 esparse -> ../esprima/bin/esparse.js
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✔ build.test                → Done (took 0 sec)
ℹ build.test                → [verbose] Build type=exec name=test is ready.
ℹ build.test2               → Building test2 (type exec) at version v-5720dec677...
ℹ build.test2               → [verbose] Done syncing sources (took 0 sec)
✖ build.test2               → Failed (took 0 sec)
✖ build.test2               → Failed processing Build type=exec name=test2 (took 0.03 sec). This is what happened:

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Build staging: Failed copying file from '/Users/steffen/repro/node_modules/.bin/acorn' to '/Users/steffen/repro/.garden/build/test/node_modules/.bin/acorn' because a directory exists at the target path
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
1 requested build action(s) failed!

See .garden/error.log for detailed error message
```

**Comment by stefreak** [comment]
@alexkuretz thanks again for reporting this! Can you have a quick look at the latest edge-bonsai version by running `garden self-update edge-bonsai` to see if it works for you now?

**Comment by stefreak** [comment]
Now also has been released as part of `0.13.41`.

**Comment by alexkuretz** [comment]
Thanks for the quick fix, we've tested with 0.13.41 and the issue is resolved
